src/traits.rs

```diff
@@ -288,6 +288,13 @@ pub trait Random: Sized {
     fn random(rng: &mut impl CryptoRngCore) -> Self;
 }
 
+/// Random number generation support.
+#[cfg(feature = "rand_core")]
+pub trait RandomBits: Sized {
+    /// Generate a cryptographically secure random value.
+    fn random_bits(rng: &mut impl CryptoRngCore, bit_length: u32, bits_precision: u32) -> Self;
```

Comment:
It seems like there are several failure cases with this API. Perhaps it should return a `Result` or `CtOption`?

In `crypto-common` we migrated to returning e.g. `rand_core::Error` and using `try_fill_bytes` so as to also handle potential RNG failures in various environments

Reply:
I don't know how much good will that do for the user. It's not something you can recover from, or pass it on to a higher level. If one of those panics triggers, it's just a plain bug in the user code.

Reply:
Returning a `Result` at least gives the caller a choice, versus always crashing the entire program.

(As it were, this crate was originally written with the goal of minimizing panics, but it seems that's gone out the window, especially with `Boxed*`)

Reply:
How about having a `try_random_bits` method that returns a result, and a provided `random_bits` method which panics?

That follows the same pattern as the [`RngCore` trait](https://docs.rs/rand_core/0.6.4/rand_core/trait.RngCore.html).

Reply:
Good idea, I'll do that. Should it actually propagate the `RngCore` error?

Reply:
That seems like a reasonable place to start, though you may need to define a separate error type to capture cases like `bit_length` > `bits_precision` (where `rand_core::Error` can be a variant like `RngError`)

Reply:
Yes, that's what I meant. Wrap the inner error and add our error cases.

Reply:
I guess you could use a custom error in the [`CUSTOM_START`](https://docs.rs/rand_core/latest/rand_core/struct.Error.html#associatedconstant.CUSTOM_START) range and the `From<NonZeroU32>` impl to construct a `rand_core::Error` if you wanted to go that route, though that may not be helpful from a usability perspective